pyk/src/pyk/cterm.py

```diff
@@ -16,9 +17,15 @@ class CTerm:
 
     def __init__(self, term: KInner) -> None:
         config, constraint = splitConfigAndConstraints(term)
-        constraints = tuple(flattenLabel('#And', constraint))
+        constraints = dedupe(flattenLabel('#And', constraint))
```

Comment:
Is `dedupe` faster or something like `list(set(...))`?

Reply:
Good point, `list(set(...))` is faster.

At this point, do we use `dedupe` over non-hashable types? If not, we can change the default implementation there so that it is as fast as `list(set(...))`, but keeps the order of elements.

Reply:
So we would continue to use `dedupe`, but it would be faster and preserve order? I would like it to preserve order, that seems like a nice property...

Reply:
Added a function `unique` for this purpose: `dedupe(xs) == list(unique(xs))`, `tuple(dedupe(xs)) == tuple(unique(xs))`, etc. We can remove / reimplement `dedupe` once we're sure we do not try to use it on collections of unhashable elements.